hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1212,10 +1223,12 @@ int MDMParser::_cbCPIN(int type, const char* buf, int len, Sim* sim)
     return WAIT;
 }
 
-int MDMParser::_cbCCID(int type, const char* buf, int len, char* ccid)
+int MDMParser::_cbCCID(int type, const char* buf, int len, CStringHelper* str)
 {
-    if ((type == TYPE_PLUS) && ccid) {
-        if (sscanf(buf, "\r\n+CCID: %[^\r]\r\n", ccid) == 1) {
+    if (str && str->str && (str->size > 1) && (type == TYPE_PLUS)) {
+        char format[32] = {};
+        snprintf(format, sizeof(format), "\r\n+CCID: %%%u[^\r]\r\n", str->size-1);
+        if (sscanf(buf, "\r\n+CCID: %[^\r]\r\n", str->str) == 1) {
```

Comment:
`format` should be used here

Reply:
My bad, that was the whole point of using format. 🤦